data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java

```diff
@@ -59,43 +60,67 @@ public SimpleDataService(
         );
     }
 
-    public Pair<AtomicLong, AtomicLong> read(final DataReaderRequest readerRequest, final OutputStream out) {
-        final AtomicLong recordsProcessed = new AtomicLong(0);
-        final AtomicLong recordsReturned = new AtomicLong(0);
+    /**
+     * Includes the resources into the OutputStream that is to be provided to the client
+     *
+     * @param readerRequestModel the information for the resources in the context of the request
+     * @param out                an {@link OutputStream} to write the stream of resources to (after applying rules)
+     * @param recordsProcessed   number of records that have been processed
+     * @param recordsReturned    number of records that have been returned
+     * @return boolean indicating that the process has been successful
+     */
+    public CompletableFuture<Boolean> read(final DataResponse readerRequestModel, final OutputStream out,
+                                           final AtomicLong recordsProcessed, final AtomicLong recordsReturned) {
 
-        LOGGER.debug("Reading from reader with request {}", readerRequest);
-        DataReaderResponse readerResponse = this.dataReader.read(readerRequest, recordsProcessed, recordsReturned);
+        return CompletableFuture.supplyAsync(() -> {
+            LOGGER.debug("Reading from reader with request {}", readerRequestModel);
+            DataReaderRequest readerRequest = new DataReaderRequest()
+                    .context(readerRequestModel.getContext())
+                    .user(readerRequestModel.getUser())
+                    .resource(readerRequestModel.getResource())
+                    .rules(readerRequestModel.getRules());
+            DataReaderResponse readerResponse = dataReader.read(readerRequest, recordsProcessed, recordsReturned);
 
-        LOGGER.debug("Writing reader response {} to output stream", readerResponse);
-        try {
-            readerResponse.getWriter().write(out);
-            out.close();
-        } catch (IOException ex) {
-            throw new ReadException(readerRequest, ex);
-        }
+            LOGGER.debug("Writing reader response {} to output stream", readerResponse);
+            try {
+                readerResponse.getWriter().write(out);
+                out.close();
+            } catch (IOException ex) {
+                throw new ReadException("Failed to write data out to the output stream.", ex);
+            }
 
-        LOGGER.debug("Output stream closed, {} processed and {} returned, auditing success with audit service", recordsProcessed.get(), recordsReturned.get());
-        return Pair.of(recordsProcessed, recordsReturned);
+            LOGGER.debug("Output stream closed, {} processed and {} returned, auditing success with audit service", recordsProcessed.get(), recordsReturned.get());
+            return true;
+        });
     }
 
+    /**
+     * Seraliser is is added to the {@link DataReader} after the application has started.
```

Comment:
```suggestion
     * Seraliser is added to the {@link DataReader} after the application has started.
```

Reply:
?? Not sure why, but this was not meant to be in this class.  The code and the comments have all been removed.  The are in the DataApplication class.